### ISSUE TITLE
Tiling rules take precedence over the window's specified tiling mode

### ISSUE THREAD

**Issue by nanzhong** [koekeishiya/chunkwm#408]
This may be intentional by design, but to me the expected behaviour feels like it should be different.

Windows matching a specific tiling rule ignores the windows overridden tiling mode.

eg. With the following tiling rule
```
chunkc tiling::rule --owner Emacs --except "^$" --state tile
```

Emacs will always be tiled even if I do
```
chunkc tiling::window --toggle float
```
on it.

My expectation here is that if the manually overridden window tiling modes should take precedence over the value set by the rules. The rules should set the "default" tiling of a window if no window tiling mode is specified.

**Comment by koekeishiya** [comment]
I'm unable to reproduce this.

Windows that are tiled due to a rule properly floats for me when I issue `chunkc tiling::window --toggle float` on it.

**Comment by nanzhong** [comment]
Ah, so I think the issue is actually triggers on resizing the window. I just noticed that when I toggle float on emacs it does a little resize jiggle, and the pops into tiled mode immediately after that.

I was also able to reproduce this with another application (that also had a similar rule, but does not do a resize jiggle when toggling float) by manually resizing the window after toggling the float.

**Comment by nanzhong** [comment]
It's also possible to reproduce by following the float toggle with a `chunkc tiling::window --grid-layout`.

eg. from my `.skhdrc`
```
# float / unfloat window
cmd + alt - t : chunkc tiling::window --toggle float

# float / unfloat window and center on screen
cmd + alt - c : chunkc tiling::window --toggle float;\
                chunkc tiling::window --grid-layout 4:4:1:1:2:2
```

`cmd + alt - t` works as expected, but any resize or move of the window will cause it to be tiled again.
`cmd + alt  - c` causes the window to float, but as soon as the `--grid-layout` command is called, the windows tiles again.

**Comment by dominiklohmann** [comment]
Does the emacs window title change when you resize it? Because that would certainly trigger the rule again.

**Comment by koekeishiya** [comment]
> Does the emacs window title change when you resize it? Because that would certainly trigger the rule again.

If this is the case, and hopefully it is, that is indeed a bug. Should be a simple problem to solve.

**Comment by koekeishiya** [comment]
Reproduced the issue and fixed on master. Thanks @dominiklohmann for mentioning the title-change thing, and thanks for catching this bug @nanzhong 

Please do confirm that it has been solved for you :)

If you install using homebrew, you can install the master branch version using `brew install chunkwm --HEAD`

**Comment by nanzhong** [comment]
@dominiklohmann you are absolutely right! Emacs shows the dimensions in the title after resize. The other application I tested with was Terminal.app, which I've configured to also show dimensions in the title bar.

@koekeishiya that does indeed solve the issue. Thanks for the incredible turn around time to fixing this!

Another potential solution is https://github.com/koekeishiya/chunkwm/pull/409. I've verified that it also works with a rule config such as `chunkc tiling::rule --owner Emacs --except "^$" --state tile --ignore-title-updates`
